rgdps/common/validators.py

```diff
@@ -28,3 +34,31 @@ def _validate(cls, value: Any, _: core_schema.ValidationInfo) -> Base64String:
             return Base64String(hashes.decode_base64(value))
         except Exception as e:
             raise ValueError(f"Input is not valid base64") from e
+
+
+class TextBoxString(str):
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls,
+        _: type[Any],
+    ) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls._validate,
+            core_schema.str_schema(),
+        )
+
+    @classmethod
+    def _validate(cls, value: Any, _: core_schema.ValidationInfo) -> TextBoxString:
+        if not isinstance(value, (str, bytes)):
+            raise TypeError("Value must be str or bytes")
+
+        if isinstance(value, bytes):
+            value = value.decode()
+
+        # Value needs to be: stripped and alphanumeric.
+        value = value.strip()
+
+        if not value.isalnum():
```

Comment:
![image](https://github.com/RealistikDash/RealistikGDPS/assets/36131887/d49e38f0-88b5-4e6e-b680-23d8f02cf4ac)
The function `str.isalnum()` does not allow spaces, which some Geometry Dash textboxes allow (such as level name).

Reply:
should be fixed in 08c940a